### ISSUE TITLE
[java] ImplicitSwitchFallThrough: False-positive with nested switch statements

### ISSUE THREAD

**Issue by sratz** [pmd/pmd#4948]
**Affects PMD Version:**  7.0.0

**Rule:** [category/java/errorprone.xml/ImplicitSwitchFallThrough](https://docs.pmd-code.org/latest/pmd_rules_java_errorprone.html#implicitswitchfallthrough)

**Description:**

**Code Sample demonstrating the issue:**

```java
public class Test {

	boolean foo(String a, String b) {

		switch (a) {
		case "a1":
			return true;
		case "a2":
			switch (b) {
			case "b1":
				return true;
			default:
				return false;
			}
		default:   // <--- false-positive ImplicitSwitchFallThrough
			return false;
		}
	}

}
```

**Outcome:**

PMD reports a violation at line 15, but that's wrong. That's a false positive.

the outer `case "a2"` cannot fall through, since the inner switch is exhaustive and returns from all branches.

**Running PMD through:** *CLI | Maven*
